src/nwbinspector/checks/_nwb_containers.py

```diff
@@ -42,6 +42,39 @@ def check_large_dataset_compression(
     return None
 
 
+@register_check(importance=Importance.CRITICAL, neurodata_type=NWBContainer)
+def check_dataset_not_empty(nwb_container: NWBContainer) -> Optional[Iterable[InspectorMessage]]:
+    """
+    Check if any datasets in the container are empty (have zero elements).
+
+    Empty datasets can cause issues with analysis and visualization tools, and generally indicate
+    missing or incomplete data.
+
+    Parameters
+    ----------
+    nwb_container: NWBContainer
+        The NWB container to check for empty datasets.
+
+    Returns
+    -------
+    Optional[Iterable[InspectorMessage]]
+        Inspector messages for each empty dataset found, or None if no empty datasets are found.
+    """
+    for field_name, field in getattr(nwb_container, "fields", dict()).items():
+        if not isinstance(field, (h5py.Dataset, zarr.Array)):
```

Comment:
I think we may also want to check for other empty array-like objects here, including numpy arrays and DataIO objects. Could use something like: 
```suggestion
        is_array_data = hasattr(value, "shape") and hasattr(value, "dtype")
        if not is_array_data:
```